event/viewtopic.php

```diff
@@ -285,7 +285,13 @@ public function show_hookup_viewtopic($event)
 			}
 
 		}
-		foreach ($this->hookup->hookup_users as $hookup_user)
+		$sorted_hookup_users = array();
+		foreach ($this->hookup->hookup_users as $user_id => $hookup_user)
+		{
+			$sorted_hookup_users[$user_details[$user_id]['username']] = $hookup_user;
+		}
+		ksort($sorted_hookup_users);
+		foreach ($sorted_hookup_users as $hookup_user)
```

Comment:
I like the idea of sorting by username. But I think it would actually be better to use the SQL command right above that is fetching the user details that you use for sorting directly, instead of using an additional loop and user array for the sorting.

So instead I would suggest to:
- [ ] Add an `order by username` to the SQL query in line 278/279.
- [ ] Use the `$user_details` array for the foreach loop instead of using the `$hookup_user` (something like `foreach ($user_details as $user_id => $single_user_details)`)
- [ ] Fill `$hookup_user` inside the foreach loop (`$hookup_user = $this->hookup->hookup_users[$user_id]`)
- [ ] Adjust the contents of the foreach loop to use `$single_user_details` instead of `$user_details` and `$user_id` instead of `$hookup_user['user_id']`

I hope the variable $user_id is not used for something else important right here, I have not checked.